File: datascience/lab1/homework/q6.R
raw <- readline("Enter a number (default 29): ")
n <- if (nchar(raw) == 0) 29 else as.integer(raw)

is_prime <- TRUE

if (n < 2) {
  is_prime <- FALSE
} else {
  i <- 2
  while (i * i <= n) {
    if (n %% i == 0) {
      is_prime <- FALSE
      break
    }
    i <- i + 1
  }
}

if (is_prime) {
  cat(sprintf("%d is a prime number.\n", n))
} else {
  cat(sprintf("%d is not a prime number.\n", n))
}

File: datascience/lab1/py/homework/q6.py
n = int(input("Enter a number (default 29): ").strip() or 29)

is_prime = True

if n < 2:
    is_prime = False
else:
    i = 2
    while i * i <= n:
        if n % i == 0:
            is_prime = False
            break
        i += 1

if is_prime:
    print(f"{n} is a prime number.")
else:
    print(f"{n} is not a prime number.")
